### PR TITLE
Firestore: Preserve reference to missing documents in 'Client.get_all'.

### LINKED ISSUE #7564: Firestore: get_all removes .id key for non existing DocumentSnapshot
When I use the `get_all` function to load documents and some are non-existing the DocumentSnapshot which are non-existing lose their `.id` key. That is unfortunate as I don't know which document lost its ID. However, it's still possible to call `.exists`.

Requesting the exact same non-existing object from the database, also return a DocumentSnapshot while this time, the `.id` can be referenced and exists.

#### Environment details

1.) Firestore
2.) Linux-4.14.104-78.84.amzn1.x86_64-x86_64-with-glibc2.9
3.)  Python 3.6.5 :: Anaconda custom (64-bit)
4.) google-cloud-firestore==0.31.0

#### Code example

```python

print(db.collection("something").document('thisDoesNotExists').get().exists)
# produces: False
print(db.collection("something").document('thisDoesNotExists').get().id)
# produces: thisDoesNotExists
print(list(db.get_all([db.collection("something").document('thisDoesNotExists')]))[0].exists)
# produces: False
print(list(db.get_all([db.collection("something").document('thisDoesNotExists')]))[0].id)
# produces: See Stack trace
```

#### Stack trace
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-218-5094e5147636> in <module>()
      2 print(db.collection("something").document('thisDoesNotExists').get().id)
      3 print(list(db.get_all([db.collection("something").document('thisDoesNotExists')]))[0].exists)
----> 4 print(list(db.get_all([db.collection("something").document('thisDoesNotExists')]))[0].id)

~/anaconda3/envs/python3/lib/python3.6/site-packages/google/cloud/firestore_v1beta1/document.py in id(self)
    592             str: The last component of the path of the document.
    593         """
--> 594         return self._reference.id
    595 
    596     @property

AttributeError: 'NoneType' object has no attribute 'id'
```

### PR DESCRIPTION
**Pull request by pchauhan-qlogic**
Fix #7564

### CODE REVIEW

File: firestore/tests/system.py [hunk 1]
```diff
@@ -768,7 +768,12 @@ def test_get_all(client, cleanup):
 
     assert snapshots[0].exists
     assert snapshots[1].exists
+
+    # missing document attribute
     assert not snapshots[2].exists
+    assert snapshots[2].id == "b"
+    assert not snapshots[2]._data
```
Comment (tseaver) [inline firestore/tests/system.py]: Please move these assertions to the unit test below/

File: firestore/tests/unit/v1/test_client.py [hunk 2]
```diff
@@ -596,8 +596,9 @@ def test_found(self):
     def test_missing(self):
         ref_string = self._dummy_ref_string()
         response_pb = _make_batch_response(missing=ref_string)
+        reference_map = {ref_string: mock.sentinel.reference}
 
-        snapshot = self._call_fut(response_pb, {})
+        snapshot = self._call_fut(response_pb, reference_map)
```
Comment (tseaver) [inline firestore/tests/unit/v1/test_client.py]: Add assertions here about the snapshot's `id`, data.

**Comment by tseaver** [comment]
The Firestore system test failure is due to #7130.

File: firestore/tests/unit/v1/test_client.py (hunk 2 above)
Comment (pchauhan-qlogic) [reply firestore/tests/unit/v1/test_client.py]: Added assertions about snapshot's id and data @tseaver

File: firestore/tests/system.py (hunk 1 above)
Comment (pchauhan-qlogic) [reply firestore/tests/system.py]: Done

**Comment by tseaver** [comment]
Unrelated Bigtable snipped flakiness.

**Pull request merged**